mage/template.go

```diff
@@ -20,8 +21,57 @@ import (
 )
 
 func main() {
-	// These functions are local variables to avoid name conflicts with 
-	// magefiles.
+	// Use local types and functions in order to avoid name conflicts with additional magefiles.
+	type arguments struct {
+		Verbose       bool          // print out log statements
+		List          bool          // print out a list of targets
+		Help          bool          // print out help for a specific target
+		IgnoreDefault bool          // ignore the default build target and print all build targets instead
+		Timeout       time.Duration // set a timeout to running the targets
+		Args          []string      // args contain the non-flag command-line arguments
+	}
+	initArgs := func() arguments {
+		args := arguments{}
+		// default flag set with ExitOnError and auto generated PrintDefaults should be sufficient
+		flag.BoolVar(&args.Verbose, "v", false, "show verbose output when running mage targets")
+		flag.BoolVar(&args.List, "l", false, "list mage targets in this directory")
+		flag.BoolVar(&args.Help, "h", false, "print out help for a specific target")
+		flag.BoolVar(&args.IgnoreDefault, "ignoredefault", false, "ignore the default build target")
+		flag.DurationVar(&args.Timeout, "t", 0, "timeout in duration parsable format (e.g. 5m30s)")
+		flag.Parse()
+		args.Args = flag.Args()
+		// With golangs flag package we can't distinguish, whether a flag's default value has been:
+		//  * provided by the user or
```

Comment:
So, the way to do the defaults to environment variables is that when you declare the default up above, instead of hard coding the default to false, get the environment variable's value, and use that as the default.